starter-code/src/clue.js

```diff
@@ -1,94 +1,152 @@
+  var weaponsArray = [];
+  weaponsArray.push(weapon1);
+  weaponsArray.push(weapon2);
+  weaponsArray.push(weapon3);
+  weaponsArray.push(weapon4);
+  weaponsArray.push(weapon5);
+  weaponsArray.push(weapon6);
+  weaponsArray.push(weapon7);
+  weaponsArray.push(weapon8);
+  weaponsArray.push(weapon9);
+  
+  
+  
+  function randomSelector (array) {
+  return array[Math.floor(Math.random() * array.length)];
+  }
+  
+  
+  function pickMistery () { 
+   var misteryEnvelope =[]
+  
+   misteryEnvelope.push(
+    randomSelector(charactersArray),    
+   randomSelector(weaponsArray),
+   randomSelector(roomsArray),
+   
+   )
+  return misteryEnvelope
 
-drOrchid
-frst_name:   Doctor
-lst_name:    Orchid
-color:        white
-description:  PhD in plant toxicology. Adopted daughter of Mr. Boddy
-age:          26
-image:        http://www.radiotimes.com/uploads/images/Original/111967.jpg
-ocupation:   Scientist
-
-profPlum
-first_name:   Victor
-last_name:    Plum
-color:        purple
-description:  Billionare video game designer
-age:          22
-image:        https://metrouk2.files.wordpress.com/2016/07/professor-plum.jpg
-occupation:   Designer
-
-missScarlet
-first_name:   Kasandra
-last_name:    Scarlet
-color:        red
-description:  She is an A-list movie star with a dark past
-age:          31
-image:        https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg
-occupation:   Actor
-
-mrsPeacock
-first_name:   Eleanor
-last_name:    Peacock
-color:        blue
-description:  She is from a wealthy family and uses her status and money to earn popularity
-age:          36
-image:        https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg
-occupation:   Socialité
-
-mrMustard
-first_name:   Jack
-last_name:    Mustard
-color:        yellow
-description:  He is a former football player who tries to get by on his former glory
-age:          62
-image:        https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg
-occupation:   Retired Football player
-
-// Weapons
-
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
-
-// Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
-
-// Characters Collection
-var charactersArray = [];
-
-// Rooms' Collection
-var roomsArray = [];
-
-// Weapons Collection
-var weaponsArray = [];
+  
+  }
+  
+  var misteryEnvelope = pickMistery();
+  function revealMistery (array) {
+    return array[0].first_name + " " + array[0].last_name + " killed Mr.Boddy using the " + array[1].name + " in the " + array[2].name + "!!!"
+  }
```

Comment:
you did everything correct, except you forgot to add four '!!!!'. In your code you had three, but jasmine was checking for four